kubectl-plugin/pkg/cmd/submit/submit_job.go

```diff
@@ -0,0 +1,582 @@
+			if !scanNotDone {
+				break
+			}
+		}
+	}()
+
+	// Wait till rayJobID is populated
+	if rayJobID == "" {
+		rayJobID = <-rayJobIDChan
+	}
+	// Add annotation to RayJob with the correct ray job id and update the CR
+	options.RayJob, err = k8sClients.DynamicClient().Resource(util.RayJobGVR).Namespace(*options.configFlags.Namespace).Get(ctx, options.RayJob.GetName(), v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to get latest version of Ray Job")
+	}
+
+	rayJobAnnotations := options.RayJob.GetAnnotations()
+	if rayJobAnnotations == nil {
+		rayJobAnnotations = make(map[string]string)
+	}
+
+	rayJobAnnotations["ray.io/ray-job-submission-id"] = rayJobID
+	options.RayJob.SetAnnotations(rayJobAnnotations)
+
+	_, err = k8sClients.DynamicClient().Resource(util.RayJobGVR).Namespace(*options.configFlags.Namespace).Update(ctx, options.RayJob, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("Error occurred when trying to add job ID to rayJob: %w", err)
+	}
+
+	// Wait for ray job submit to finish.
+	err = cmd.Wait()
+	if err != nil {
+		return fmt.Errorf("Error occurred with ray job submit: %w", err)
+	}
+	return nil
+}
+
+func (options *SubmitJobOptions) raySubmitCmd() ([]string, error) {
+	raySubmitCmd := []string{"ray", "job", "submit", "--address", dashboardAddr}
+
+	if len(options.runtimeEnv) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--runtime-env", options.runtimeEnv)
+	}
+	if len(options.runtimeEnvJson) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--runtime-env-json", options.runtimeEnvJson)
+	}
+	if len(options.submissionID) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--submission-id", options.submissionID)
+	}
+	if options.entryPointCPU > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--entrypoint-num-cpus", fmt.Sprintf("%f", options.entryPointCPU))
+	}
+	if options.entryPointGPU > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--entrypoint-num-gpus", fmt.Sprintf("%f", options.entryPointGPU))
+	}
+	if options.entryPointMemory > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--entrypoint-memory", fmt.Sprintf("%d", options.entryPointMemory))
+	}
+	if len(options.entryPointResource) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--entrypoint-resource", options.entryPointResource)
+	}
+	if len(options.metadataJson) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--metadata-json", options.metadataJson)
+	}
+	if options.noWait {
+		raySubmitCmd = append(raySubmitCmd, "--no-wait")
+	}
+	if len(options.headers) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--headers", options.headers)
+	}
+	if len(options.verify) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--verify", options.verify)
+	}
+	if len(options.logStyle) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--log-style", options.logStyle)
+	}
+	if len(options.logColor) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--log-color", options.logColor)
+	}
+	if len(options.workingDir) > 0 {
+		raySubmitCmd = append(raySubmitCmd, "--working-dir", filepath.Clean(options.workingDir))
```

Comment:
I changed the location of filepath clean to end of validation for `working-dir` since it returns `.` for empty path/string and it's unknown if that is an actual user input or not. And if the working-dir is from inside the runtime-env, it would be difficult to know whether to override or not.